codegen/src/test/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/JsonParserGeneratorTest.kt

```diff
@@ -125,39 +125,47 @@ class JsonParserGeneratorTest {
-                // ${writer.format(documentGenerator)}
-                // ${writer.format(payloadGenerator)}
-
-                let json = br#"
-                    { "top":
-                        { "extra": 45,
-                          "field": "something",
-                          "choice":
-                              { "int": 5 },
-                          "empty": { "not_empty": true }}}
-                "#;
-
-                let output = ${writer.format(operationGenerator!!)}(json, output::op_output::Builder::default()).unwrap().build();
-                let top = output.top.expect("top");
-                assert_eq!(Some(45), top.extra);
-                assert_eq!(Some("something".to_string()), top.field);
-                assert_eq!(Some(Choice::Int(5)), top.choice);
-                let output = ${writer.format(operationGenerator!!)}(b"", output::op_output::Builder::default()).unwrap().build();
-                assert_eq!(output.top, None);
-
-
-                // empty error
-                let error_output = ${writer.format(errorParser!!)}(b"", error::error::Builder::default()).unwrap().build();
-                assert_eq!(error_output.message, None);
-
-                // error with message
-                let error_output = ${writer.format(errorParser!!)}(br#"{"message": "hello"}"#, error::error::Builder::default()).unwrap().build();
-                assert_eq!(error_output.message.expect("message should be set"), "hello");
-                """
+                test = """
+                                use model::Choice;
+
+                                // Generate the document serializer even though it's not tested directly
+                                // ${writer.format(documentGenerator)}
+                                // ${writer.format(payloadGenerator)}
+
+                                let json = br#"
+                                    { "top":
+                                        { "extra": 45,
+                                          "field": "something",
+                                          "choice": { "int": 5 },
+                                          "empty": { "not_empty": true }
+                                        }
+                                    }
+                                "#;
+
+                                let output = ${writer.format(operationGenerator!!)}(json, output::op_output::Builder::default()).unwrap().build();
+                                let top = output.top.expect("top");
+                                assert_eq!(Some(45), top.extra);
+                                assert_eq!(Some("something".to_string()), top.field);
+                                assert_eq!(Some(Choice::Int(5)), top.choice);
+
+                                // empty body
+                                let output = ${writer.format(operationGenerator)}(b"", output::op_output::Builder::default()).unwrap().build();
```

Comment:
Hmmm... when I removed these bang operators in these tests before, it caused the output to be formatted incorrectly such that it wouldn't compile. Did something change since then that makes this work now?

Reply:
yeah, the parsers aren't nullable anymore